### PR TITLE
Add "consensus mechanisms" docs section

### PR DESCRIPTION
**Pull request by samajammin**
## Description

Add "consensus mechanisms" docs section

## Related Issue
#1438, specifically this comment: https://github.com/ethereum/ethereum-org-website/issues/1438#issuecomment-699212075

## Additional context

Some potential sources of inspiration:

**Proof of stake**
- https://docs.ethhub.io/ethereum-roadmap/ethereum-2.0/proof-of-stake/ (CC license)
- https://eth.wiki/en/concepts/proof-of-stake-faqs (from GH wiki, which has a CC license)
- https://consensys.net/blog/blockchain-explained/what-is-proof-of-stake/

**Proof of work**
- https://en.wikipedia.org/wiki/Proof_of_work
- https://en.bitcoin.it/wiki/Proof_of_work

**Mining**
As a beginner I want... 
- an overview of what mining is so that I understand its place in Ethereum
- to know why mining is necessary
    - how it keeps ethereum secure
    - how it ensures decentralization
- to know its relationship with fees
- https://ethereum.org/en/whitepaper/#mining
- https://eth.wiki/fundamentals/mining
- https://docs.ethhub.io/using-ethereum/mining/

### CODE REVIEW

**Comment by samajammin** [comment]
@ryancreatescopy curious to get your thoughts on this approach.

In my mind, there's a few decisions to make:
- Overall, does this structure make the most sense?
- If yes, should this entire section be within "Foundational topics"? An alternative approach might be to keep that section as is, with just a link to the "Mining" page. We could then add the "consensus protocols" section under "Advanced".
- Terminology: 
  - "consensus mechanisms" vs. "consensus protocols"? They appear to have about the same search volume ([Google Adwords screenshot](https://share.getcloudapp.com/llu2QqYe)). I think I favor "consensus mechanisms" since we use the term "protocol" elsewhere (e.g. the Ethereum protocol, DeFi protocols)
  - "proof-of-work" vs. "Proof of Work"? Which should we role out consistently (I prefer "Proof of Work (PoW)")

File: src/content/developers/docs/consensus-protocols/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: Consensus protocols
```
Comment (ryancreatescopy) [inline src/content/developers/docs/consensus-protocols/index.md:2]: I agree with your comment that we should go with consensus mechanisms so as not to confuse with protocols.

File: src/content/developers/docs/consensus-protocols/index.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+## Types of consensus protocols
+
+<!-- TODO -->
+<!-- Why do different consensus protocols exist? -->
+<!-- What are the tradeoffs of each? -->
+
+### Proof of work
+
+Ethereum, like Bitcoin, currently uses a [proof-of-work (PoW)](/developers/docs/consensus-protocols/pow/) consensus protocol.
```
Comment (ryancreatescopy) [inline src/content/developers/docs/consensus-protocols/index.md:45]: A rule on hyphenation.

Hyphenate proof-of-work when used in the middle of a sentence.
You don't need to hyphenate it when it comes at the end of a sentence like proof of work. 
It also doesn't need to be hyphenated in a title or when used in isolation like in a list.

File: src/content/developers/docs/consensus-protocols/pos/index.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+incomplete: true
+---
+
+## Prerequisites {#prerequisites}
+
+To better understand this page, we recommend you first read up on [consensus mechanisms](/developers/docs/consensus-protocols/).
+
+## What is Proof of Stake (PoS)?
+
+Proof of stake is a type of [consensus mechanism](/developers/docs/consensus-protocols/) used by blockchain networks to achieve distributed consensus. For Ethereum, the purpose of Proof of Stake is the same as [Proof of Work](/developers/docs/consensus-protocols/pow/) - to ensure that all nodes agree the state of the network and on the order of transactions.
```
Comment (ryancreatescopy) [inline src/content/developers/docs/consensus-protocols/pos/index.md:15]: hyphenate pos and pow here

File: src/content/developers/docs/consensus-protocols/pos/index.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Proof of Stake comes with a number of improvements compared to [Proof of Work](/developers/docs/consensus-protocols/pow/), including: energy efficiency, lower barriers to entry, stronger crypto-economic incentives.
```
Comment (ryancreatescopy) [inline src/content/developers/docs/consensus-protocols/pos/index.md:17]: hyphenate and I'm not sure we need to capitalise Stake and Work in these instances.

File: src/content/developers/docs/consensus-protocols/pow/index.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+The Proof of Work consensus mechanism is currently the most widely-used consensus mechanism. It was pioneered by Satoshi Nakamoto with the release of Bitcoin in 2008 and it has powered the majority of high-profile blockchains, including Ethereum.
+
+## How does it work?
+
+<!-- TODO how does it work -->
+
+## What is mining?
+
+Mining is the lifeblood of Proof of Work. It is the process of creating a block transactions to be added to the Ethereum blockchain.
```
Comment (ryancreatescopy) [inline src/content/developers/docs/consensus-protocols/pow/index.md:27]: Feel like the second sentence might have some typos. Should it be "block of transactions"?

File: src/content/developers/docs/consensus-protocols/pow/index.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+Miners operate by solving computationally difficult puzzles in order to produce blocks, which serves as a way to secure the network from attacks. In return, they receive a reward in the form of newly minted ETH.
```
Comment (ryancreatescopy) [inline src/content/developers/docs/consensus-protocols/pow/index.md:29]: Could we add a comment about how this process secures the network, e.g. the investment in resources required, the fact that you'd have to coerce 51% of the other miners into mining your fraudulent blocks etc.

**Review by ryancreatescopy — CHANGES_REQUESTED** [review]

File: src/content/developers/docs/consensus-protocols/index.md (hunk 2 above)
Comment (samajammin) [reply src/content/developers/docs/consensus-protocols/index.md]: Interesting! Good to know.

There also appears to be different naming styles across the internet.

e.g. https://consensys.net/blog/blockchain-explained/what-is-proof-of-stake/
Here they always capitalize (Proof of Work & Proof of Stake)

e.g. https://en.wikipedia.org/wiki/Proof_of_stake
Seems to keep things lowercase

I default to you on this, just want us to be consistent.

File: src/content/developers/docs/consensus-protocols/pow/index.md (hunk 6 above)
Comment (samajammin) [reply src/content/developers/docs/consensus-protocols/pow/index.md]: Ya totally - the content in this PR if very much incomplete, I mainly wanted early feedback on the page structure. 

I welcome any additions tho 😄

**Comment by ryancreatescopy** [comment]
@samajammin this is ready for review

**Review by ryancreatescopy — APPROVED** [review]

**Comment by fulldecent** [comment]
Recommending to close this issue.

The published page is at https://ethereum.org/en/developers/docs/consensus-mechanisms/, it has a call for help which is more visible than this issue and it already briefly explains PoW and PoS.

I would say this minimum viable product required by this issue has been satisfied.

**Pull request merged**